homeassistant/components/light/wemo.py

```diff
@@ -140,3 +148,89 @@ def turn_off(self, **kwargs):
     def update(self):
         """Synchronize state with bridge."""
         self.update_lights(no_throttle=True)
+
+
+class WemoDimmer(Light):
+    """Representation of a WeMo dimmer."""
+
+    def __init__(self, device):
+        """Initialize the WeMo dimmer."""
+        self.wemo = device
+        self._brightness = None
+        self._state = None
+        # look up model name once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+
+        wemo = get_component('wemo')
+        wemo.SUBSCRIPTION_REGISTRY.register(self.wemo)
+        wemo.SUBSCRIPTION_REGISTRY.on(self.wemo, None, self._update_callback)
```

Comment:
Move the registration of the callback to the coroutine `async_added_to_hass`, to avoid a race where the entity hasn't been added to home assistant before calling the callback.